docs(common errors): added new section (#2605)
nestjs/docs.nestjs.com: content/faq/errors.md

<figure><img src="/assets/injector_logs.png" /></figure>

In the above image, the string in yellow is the host class of the dependency being injected, the string in blue is the name of the injected dependency, or its injection token, and the string in purple is the module in which the dependency is being searched for. Using this, you can usually trace back the dependency resolution for what's happening and why you're getting dependency injection problems.

#### "File change detected" loops endlessly

Windows users who are using TypeScript version 4.9 and up may encounter this problem.

> This happens when you're trying to run your application in watch mode, e.g `npm run start:dev` and see an endless loop of the log messages:
> ```bash

Reviewer: ```suggestion
This happens when you're trying to run your application in watch mode, e.g `npm run start:dev` and see an endless loop of the log messages:

```bash
```

XX:XX:XX AM - File change detected. Starting incremental compilation...
XX:XX:XX AM - Found 0 errors. Watching for file changes.
```

> When you're using the NestJS CLI to start your application in watch mode it is done by calling `tsc --watch`, and as of version 4.9 of TypeScript, a [new method](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#file-watching-now-uses-file-system-events) for detecting file changes is used which is the cause of this problem.

Reviewer 2: ```suggestion
When you're using the NestJS CLI to start your application in watch mode it is done by calling `tsc --watch`, and as of version 4.9 of TypeScript, a [new strategy](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#file-watching-now-uses-file-system-events) for detecting file changes is used which is the likely to be the cause of this problem.
```

Author: Added these changes with a small grammar error fix

> When you're using the NestJS CLI to start your application in watch mode it is done by calling `tsc --watch`, and as of version 4.9 of TypeScript, a [new strategy](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#file-watching-now-uses-file-system-events) for detecting file changes is used which is likely to be the cause of this problem.

Reviewer: ```suggestion

When you're using the NestJS CLI to start your application in watch mode it is done by calling `tsc --watch`, and as of version 4.9 of TypeScript, a [new strategy](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#file-watching-now-uses-file-system-events) for detecting file changes is used which is likely to be the cause of this problem.
```

> In order to fix this problem, you need to add a setting to your tsconfig.json file after the `"compilerOptions"` option as follows:

Reviewer 2: ```suggestion
In order to fix that problem, you could try to use the old strategy by setting the [`watchOptions.watchFile` option](https://www.typescriptlang.org/tsconfig#watch-watchFile) to `"fixedPollingInterval"` as follows:
```

> In order to fix this problem, you need to add a setting to your tsconfig.json file after the `"compilerOptions"` option as follows:

Reviewer: ```suggestion
In order to fix this problem, you need to add a setting to your tsconfig.json file after the `"compilerOptions"` option as follows:

```bash
```

```bash
  "watchOptions": {
    "watchFile": "fixedPollingInterval"
  }
```

> This tells TypeScript to use the polling method for checking for file changes instead of file system events (the new default method), which can cause issues on some machines.

Reviewer: ```suggestion

This tells TypeScript to use the polling method for checking for file changes instead of file system events (the new default method), which can cause issues on some machines.
```

> You can read more about the `"watchFile"` options you can use in [this link](https://www.typescriptlang.org/tsconfig#watch-watchDirectory).

Reviewer 2: ```suggestion
You can read more about the `"watchFile"` option you can use in [TypeScript documentation](https://www.typescriptlang.org/tsconfig#watch-watchDirectory).
```
